src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,784 @@
+        if input_latent is not None:
+            if not torch.is_tensor(input_latent):
+                raise ValueError("`input_latent` must be a torch.FloatTensor.")
+            if not input_latent.dim() != 4:
+                raise ValueError(f"`input_latent` has unsupported dimensions or shape: {input_latent.shape}.")
+
+            if processing_resolution > 0:
+                max_orig = max(H, W)
+                new_H = H * processing_resolution // max_orig
+                new_W = W * processing_resolution // max_orig
+                if new_H == 0 or new_W == 0:
+                    raise ValueError(f"Extreme aspect ratio of the input image: [{W} x {H}]")
+                W, H = new_W, new_H
+            w = (W + self.latent_size_scale - 1) // self.latent_size_scale
+            h = (H + self.latent_size_scale - 1) // self.latent_size_scale
+            shape_expected = (num_images * ensemble_size, self.latent_space_size, h, w)
+
+            if input_latent.shape != shape_expected:
+                raise ValueError(
+                    f"`input_latent` has unexpected shape={input_latent.shape} expected={shape_expected}."
+                )
+
+        # generator checks
+        if generator is not None:
+            device = self._execution_device
+            if isinstance(generator, torch.Generator):
+                if generator.device != device:
+                    raise ValueError("`generator` device differs from the pipeline's device.")
+            elif isinstance(generator, list):
+                if len(generator) != num_images * ensemble_size:
+                    raise ValueError(
+                        "The number generators must match the total number of ensemble members for all input images."
+                    )
+                if not all(g.device == device for g in generator):
+                    raise ValueError("At least one of the `generator` devices differs from the pipeline's device.")
+            else:
+                raise ValueError(f"Unsupported generator type: {type(generator)}.")
+
+    @torch.no_grad()
+    @replace_example_docstring(EXAMPLE_DOC_STRING)
+    def __call__(
+        self,
+        input_image: Union[Image.Image, np.ndarray, torch.FloatTensor],
+        denoising_steps: Optional[int] = None,
+        ensemble_size: int = 1,
+        processing_resolution: Optional[int] = None,
+        match_input_resolution: bool = True,
+        resample_method_input: str = "bilinear",
+        resample_method_output: str = "bilinear",
+        batch_size: int = 1,
+        check_input: bool = True,
+        ensembling_kwargs: Optional[Dict[str, Any]] = None,
+        input_latent: Optional[Union[torch.FloatTensor, List[torch.FloatTensor]]] = None,
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None,
+        output_prediction_format: str = "np",
+        output_visualization: bool = True,
+        output_visualization_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
```suggestion
```

I think we do not need to output visualization from the pipeline, user can get it visualization with this API
```python
from diffusers.utils import visualize_depth

pipe = MarigoldDepthPipeline.from_pretrained(...)
out = pipe(image).image
visualize_depth(out).save("einstein_depth.png")
```

Reply:
Yeah agreed. We do this for our video pipelines too.

Reply:
Done